### PR TITLE
Fix update product price

### PR DESCRIPTION
**Pull request by alianza**
Fixed the updating of the product price when choosing variants. Now works with multiple variations. Testable on for instance the 'Hoodie'. Colors and sizes have different prices

### CODE REVIEW

**Comment by alianza** [comment]
⚡️ [Lighthouse report](https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1626255247502-75261.report.html) for the changes in this PR:
| Category | Score |
| --- | --- |
| 🟠 Performance | 78 |
| 🟢 Accessibility | 100 |
| 🟢 Best practices | 100 |
| 🟢 SEO | 92 |
| 🟢 PWA | 92 |
 
*Lighthouse ran on [https://headless-commerce-poc-63evnaeww-alianza.vercel.app/](https://headless-commerce-poc-63evnaeww-alianza.vercel.app/)*

**Comment by alianza** [comment]
> @alianza what wasn't working before? If I change the size or color of the hoodie, the price changes on the current site as well.
> 
> variants

It was working fine before. Though the implementation was overkill. This PR removed a lot of code but achieves the same functionality!

File: components/product/ProductView/ProductView.tsx [hunk 1]
```diff
@@ -4,72 +4,36 @@ import { NextSeo } from 'next-seo'
...
-      }))
-    })
-  }, [product.variants])
-
   const variant = getProductVariant(product, selectedOptions)
 
+  const { price } = usePrice({
+    amount: variant?.price || 0,
+    baseAmount: product.price.retailPrice,
+    currencyCode: product.price.currencyCode || 'Eur',
```
Comment (pascalw) [inline components/product/ProductView/ProductView.tsx:34]: Why is this default here? These kinds of defaults should not be hardcoded IMO. This data should come from the backend, the frontend should not make any assumptions about this.

Comment (alianza) [reply components/product/ProductView/ProductView.tsx:34]: I will investigate this. I know these constants are defined somewhere but I remember having trouble using those here.

**Review by pascalw — APPROVED** [review]
Clear. Nice cleanup!

**Pull request merged**